cmd/kubectl/kubectl.go

```diff
@@ -26,6 +29,13 @@ import (
 )
 
 func main() {
+	// We need to manually parse the arguments looking for verbosity flag and
+	// set appropriate level here, because in the normal flow the flag parsing,
+	// including the logging verbosity, happens inside cli.RunNoErrOutput.
+	// Doing it here ensures we can continue using klog during kubectl command
+	// construction, which includes handling plugins and parsing .kuberc file,
+	// for example.
+	logs.GlogSetter(cmd.GetLogVerbosity(os.Args))
```

Comment:
I think, this piece is much nicer comparing to my PoC looking workaround solution. More importantly, we don't need to call `ParseFlags` which I was worried about that can cause real headaches for us.

Besides this should be idempotent, because os.Args will be parsed to exact same log level in command execution anyway. Only real issue would be `GetLogVerbosity` has a bug, so that t`GetLogVerbosity` may set the incorrect log verbosity level. But it would be very unlikely.

Reply:
> Besides this should be idempotent, because os.Args will be parsed to exact same log level in command execution anyway. Only real issue would be `GetLogVerbosity` has a bug, so that t`GetLogVerbosity` may set the incorrect log verbosity level. But it would be very unlikely.

Even if so, that will only apply to the command construction, and the rest of the commands will get the correct logic, b/c proper flag parsing will happen when running commands - tested it :sweat_smile: . So the surface is relatively small and if that happens we should be able to quickly fix it.

Reply:
yes, I agree. Thank you.